### DISCUSSION TITLE
How to Calibrate Realsense Camera in ROS2 Humble and Apply to camera_info Topic?

### DISCUSSION THREAD

**Discussion by sakamakiy1234** [realsenseai/realsense-ros#3163] [Q&A]
Hi,

I'm currently working on a project where I'm using a Realsense camera (D457) with ROS2 Humble. I would like to calibrate the camera and apply the calibration to the camera_info topic.

I was thinking of using the ros-humble-camera-calibration package to generate a camera.yaml file and then apply this calibration. However, I couldn't find any documentation or resources on how to do this in the GitHub documentation or through web searches.

Additionally, I'm curious if the lack of documentation is because the default calibration values are already sufficient and there is no need for further calibration.

Could anyone provide guidance or point me to the right resources for calibrating the Realsense camera and applying the calibration to the camera_info topic in ROS2 Humble? 

Thank you!

**Comment by MartyG-RealSense** [comment]
Hi @sakamakiy1234  RealSense cameras are calibrated in the factory and usually already have a very good calibration after purchase.  

Are you using your D457 in 'USB' mode so that it is detected as a D455, please?  Or are you using it in MIPI mode with a GMSL cable?

You can reset the camera to its default factory-new calibration in the RealSense Viewer tool using instructions at https://github.com/IntelRealSense/librealsense/issues/10182#issuecomment-1019854487

If you need to perform a calibration then I recommend using one of the official RealSense calibration tools, as a tool not created by Intel will not be able to save a calibration permanently to the camera hardware.

The RealSense Viewer has a built-in calibration tool called **On-Chip Calibration**.  This can be found in the drop-down menu of the **More** option near the top of the Viewer's side-panel.

https://dev.intelrealsense.com/docs/self-calibration-for-depth-cameras

**Reply by sakamakiy1234** [reply]
Hi,@MartyG-RealSense

Thank you for the guidance on using the RealSense Viewer tool for calibration.

 > Are you using your D457 in 'USB' mode so that it is detected as a D455, please? Or are you using it in MIPI mode with a GMSL cable?

I am using my RealSense D457 in USB3.2 mode, and it is detected as a D455 in the RealSense Viewer. My concern is whether the calibration values are accurate for the D457 model when it is recognized as a D455.

Is the calibration performed and stored by the RealSense Viewer tool appropriate for the D457 even though it shows up as a D455?

Thank you!

**Comment by MartyG-RealSense** [comment]
Inside the casing of a D457 camera, it has the same camera 'depth module' circuit board that a D455 has.  So its calibration is indeed comparable to a D455.

A D457 operating in USB mode and detected as D455 has the same feature-set and performance as a D455.

If the camera's operating mode was changed to 'MIPI' and used with a GMSL cable instead of a USB cable, the calibration would still be the same as when it was operating as a D455 in USB mode.

**Reply by sakamakiy1234** [reply]
Thank you, @MartyG-RealSense 
I am relieved to know that the calibration for the D457 in USB mode is accurate and comparable to the D455.

**Comment by MartyG-RealSense** [comment]
You are very welcome. I'm pleased that I could help!
